### ISSUE TITLE
Subsetting rows based string columns

### ISSUE THREAD

**Issue by AdrianAntico** [h2oai/datatable#3145]
- Describe what feature you would like to see implemented.
Not sure if there is a way to do this, but I didn't see any string examples in the docs for subsetting rows. 

In R data.table, I can do: 
data <- data[StringCol %chin% c("A", "B", "C")] or data <- data[StringCol %in% c("A", "B", "C")]

I'm hoping in Python datatable I can also do something similar, such as:
data = data[StringCol in ['A', 'B', 'C'], :]
but it doesn't appear so. 

I was able to do what I wanted with this:
data = data[(f.Region == 'A') | (f.Region == 'B'), :]
but that is not scalable for broader subsetting tasks

**Comment by st-pasha** [comment]
I'm not sure what you mean by "broader subsetting tasks", could you please elaborate on this?
The examples that you gave so far are correct:
```
data[f.Region in ['A', 'B', 'C'], :]
```
is equivalent to
```
data[(f.Region == 'A') | (f.Region == 'B') | (f.Region == 'C'), :]
```

**Comment by AdrianAntico** [comment]
@st-pasha I'm getting an error message when I try to run the first bit of code.

```
>>> data = data[f.Region in ['A', 'B', 'C'], :]
TypeError: Expression f.Region == 'A' cannot be cast to bool.

You may be seeing this error because either:
  * you tried to use chained inequality such as
        0 < f.A < 100
    If so please rewrite it as
        (0 < f.A) & (f.A < 100)

  * you used keywords and/or, for example
        f.A < 0 or f.B >= 1
    If so then replace keywords with operators & or |:
        (f.A < 0) | (f.B >= 1)
    Be mindful that & / | have higher precedence than and
    or or, so make sure to use parentheses appropriately.

  * you used expression in the if statement, for example:
        f.A if f.A > 0 else -f.A
    You may write this as a ternary operator instead:
        (f.A > 0) & f.A | -f.A

  * you explicitly cast the expression into bool:
        bool(f.B)
    this can be replaced with an explicit comparison operator:
        f.B != 0
```

**Comment by st-pasha** [comment]
Yes, because `in` operator is not implemented yet (see issue #699)

**Comment by AdrianAntico** [comment]
@st-pasha okay, thanks. I'll use the functools approach for now.
